src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -795,9 +792,9 @@ public sealed override ImmutableArray<FieldSymbol> TupleElements
             }
 
             elementsMatchedByFields.Free();
+            this.TupleData!.SetUnwrappedFieldsToIndexMap(unwrappedFieldsToIndexMap);
```

Comment:
>this.TupleData!.SetUnwrappedFieldsToIndexMap(unwrappedFieldsToIndexMap); [](start = 12, length = 72)

Could we end up in a situation when member list comes from one thread and the map from another, effectively based on different sets of symbols?

Reply:
Consider if we could base this on original definitions which could be compared by reference.

---
In reply to: [426932926](https://github.com/dotnet/roslyn/pull/44231#discussion_r426932926) [](ancestors = 426932926)

Reply:
Also, if we go with just definitions, then, at least for some tuple types we could calculate this map just for the definition type and delegate to it from non-definition types. 

---
In reply to: [426933364](https://github.com/dotnet/roslyn/pull/44231#discussion_r426933364) [](ancestors = 426933364,426932926)